crates/iceberg/src/lib.rs

```diff
@@ -50,6 +50,87 @@
 //!     Ok(())
 //! }
 //! ```
+//!
+//! ## Fast append data to table
+//!
+//! ```rust, no_run
```

Comment:
This test only use `memory` file io and `memory` catalog, it should be able to run?

Reply:
We should create table here to make it run.

Reply:
I personally believe it's more valuable. Is the blocker here that we don't support creating tables yet?

Reply:
I find that the memory catalog can't update now. So we should merge #1002 first.

Reply:
Yes, I hope we can have more high-level api examples. I think this is an example of low-level api which may be useful for computing engines integrated with iceberg-rust.🤔

Reply:
> This is an example showing in crate front page, which I don't think is appropriate.

Looks reasonable to me. What about also place it at https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/writer/mod.rs

Reply:
> You mean the whole example? I'm not a big fan of that way. For writer doc, it should only focus on how to use a writer. The whole story of using different picecs in a compute engine is an advanced topic, and deserves a single chapter for that. It involves not only code, but also explaination the design in it.

Sounds reasonable for me.